alerting/src/main/kotlin/org/opensearch/alerting/settings/AlertingSettings.kt

```diff
@@ -50,6 +52,22 @@ class AlertingSettings {
             Setting.Property.NodeScope, Setting.Property.Dynamic
         )
 
+        /** Setting to help timebox doc level monitor fanout
+         */
+        val DOC_LEVEL_MONITOR_FANOUT_MAX_DURATION = Setting.positiveTimeSetting(
+            "plugins.alerting.monitor.doc_level_monitor_fanout_max_duration",
+            TimeValue.timeValueMinutes(DEFAULT_MAX_DOC_LEVEL_MONITOR_FANOUT_MAX_DURATION_MINUTES),
+            Setting.Property.NodeScope, Setting.Property.Dynamic
+        )
+
+        /** Setting to help timebox doc level monitor execution
```

Comment:
Not a blocking comment; just adding this here for visibility.

My understanding is that if an execution times out before processing all of the docs, the leftover docs should be processed by the next execution. We should mention in troubleshooting documentation that if it seems like monitor executions are consistently timing out before processing all docs, the cluster may need to be scaled up/out.